util/git/client.go

```diff
@@ -737,11 +737,11 @@ func (m *nativeGitClient) CommitSHA() (string, error) {
 
 // returns the meta-data for the commit
 func (m *nativeGitClient) RevisionMetadata(revision string) (*RevisionMetadata, error) {
-	out, err := m.runCmd("show", "-s", "--format=%an <%ae>|%at|%B", revision)
+	out, err := m.runCmd("show", "-s", "--format=%an <%ae>%n%at%n%B", revision)
```

Comment:
May be a silly question, but... are newlines allowed in a git username? 0_o

Reply:
That doesn't seem to be the case, we debated that [here](https://github.com/argoproj/argo-cd/pull/20040#discussion_r1769545321)

Also added a [test](https://github.com/argoproj/argo-cd/pull/20040/files#diff-a7b72b46e16649a80020d533dac6c2c4ed5bbc2dc72852f43a44e01037c48b55R317) for that, and it seems like Git strips the newlines from the [username](https://github.com/argoproj/argo-cd/pull/20040/files#diff-a7b72b46e16649a80020d533dac6c2c4ed5bbc2dc72852f43a44e01037c48b55R334).